src/components/OnyxProvider.js

```diff
@@ -7,7 +7,6 @@ import ComposeProviders from './ComposeProviders';
 // Set up any providers for individual keys. This should only be used in cases where many components will subscribe to
 // the same key (e.g. FlatList renderItem components)
 const [withNetwork, NetworkProvider] = createOnyxContext(ONYXKEYS.NETWORK);
-const [withPersonalDetails, PersonalDetailsProvider] = createOnyxContext(ONYXKEYS.PERSONAL_DETAILS);
```

Comment:
Ah sorry this might be an annoying realization, but by moving this out of the `OnyxProvider` we are likely introducing a performance regression. I would actually suggest that we:

- Put this back how it was so we can leverage the Context API
- Create an HOC like withCurrentUserPersonalDetails
- Have that HOC use withPersonalDetails (context API version) and just pass the currentUserPersonalDetails

Let me know if that makes sense!

Reply:
No worries! That does make sense. I can do that.